js/ui/dnd.js

```diff
@@ -622,6 +622,9 @@ var _Draggable = new Lang.Class({
         dragActor.disconnect(this._dragActorDestroyId);
         this._dragActorDestroyId = 0;
 
+        this.emit('drag-end', eventTime, false);
```

Comment:
Why does "JavaScript [abort] the execution of the function" without this change?

Reply:
Because it calls `this._dragActor.destroy()` before calling into `_dragComplete()` (that calls `Shell.util_set_hidden_from_pick(this._dragActor, false);`)

Reply:
Updated the commit message to explain that.

Reply:
I still don't quite understand why this aborts the execution of this function – does one of those calls raise an unhandled exception?

Reply:
Because after destroying the actor, the actor is invalid. Thus, calling `Shell.util_set_hidden_from_pick(this._dragActor, false);` tries to access GObject data, which fails.

Reply:
If you need a more convincing argument, it stops executing the function with the following stack:

```
nov 27 11:10:13 endless gnome-shell[1604]: Object .Gjs_BaseIcon (0x5572636c7fd0), has been already deallocated - impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: == Stack trace for context 0x5572626bf4c0 ==
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #0 0x557262ad00e0 i   resource:///org/gnome/shell/ui/dnd.js:638 (0x7f543b566c48 @ 44)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #1 0x7ffd681dfbe0 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #2 0x557262ad0058 i   resource:///org/gnome/shell/ui/dnd.js:614 (0x7f543b566b38 @ 63)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #3 0x7ffd681e0f20 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #4 0x557262acffc0 i   resource:///org/gnome/shell/ui/dnd.js:633 (0x7f543b566bc0 @ 234)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #5 0x7ffd681e2260 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #6 0x557262acff38 i   resource:///org/gnome/shell/ui/tweener.js:91 (0x7f544c0c8de0 @ 37)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #7 0x557262acfe98 i   resource:///org/gnome/gjs/modules/tweener/tweener.js:208 (0x7f544c0d3918 @ 54)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #8 0x7ffd681e4460 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:337 (0x7f544c0d39a0 @ 1626)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #9 0x7ffd681e5250 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:594 (0x7f544c0d3cd0 @ 1447)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #10 0x557262acfba8 i   resource:///org/gnome/gjs/modules/tweener/tweener.js:603 (0x7f544c0d3d58 @ 17)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #11 0x557262acfb20 i   resource:///org/gnome/shell/ui/tweener.js:40 (0x7f544c0c8ab0 @ 38)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #12 0x557262acfa70 i   resource:///org/gnome/shell/ui/dnd.js:605 (0x7f543b566ab0 @ 202)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #13 0x7ffd681e6630 b   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #14 0x557262acf9c0 i   resource:///org/gnome/shell/ui/dnd.js:567 (0x7f543b5669a0 @ 463)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #15 0x557262acf918 i   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #16 0x7ffd681e8030 b   resource:///org/gnome/shell/ui/dnd.js:514 (0x7f543b566890 @ 1042)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #17 0x557262acf788 i   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #18 0x557262acf6f0 i   resource:///org/gnome/shell/ui/dnd.js:211 (0x7f543b566450 @ 158)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #19 0x557262acf648 i   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f544c0b5de0 @ 71)
nov 27 11:10:13 endless org.gnome.Shell.desktop[1604]: #20 0x557262acf5c0 i   resource:///org/gnome/shell/ui/dnd.js:54 (0x7f543b55ccd0 @ 18)
```

Line 638 is `Shell.util_set_hidden_from_pick(this._dragActor, false);`

Reply:
OK, I believe you! :)